# Gikomba Market Stall Analysis

This notebook analyses Wanjiku's fresh-produce stall sales and stock data to understand business performance, sales growth, daily performance, and restocking needs.

In [27]:
daily_sales = [2300, 1950, 2800, 2100, 3600, 5200, 4100,
               2500, 2200, 3100, 2400, 3900, 5600, 4700]   # 14 days, KES

days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"] * 2

# qty = kg in stock, reorder = reorder level (kg), cost/price = KES per kg
stock = {
    "Tomatoes": {"qty": 18, "reorder": 25, "cost": 60, "price": 90},
    "Onions":   {"qty": 40, "reorder": 20, "cost": 70, "price": 100},
    "Sukuma":   {"qty": 9,  "reorder": 15, "cost": 30, "price": 50},
    "Potatoes": {"qty": 55, "reorder": 30, "cost": 50, "price": 75},
    "Avocado":  {"qty": 12, "reorder": 10, "cost": 80, "price": 140},
}

## Task 1 — Overall Sales Performance

I will calculate the total, average, highest, and lowest sales for the 14 days. I will also use the position of the highest and lowest sales to identify the corresponding day and week.

In [28]:
# Calculate the basic sales statistics
total_sales = sum(daily_sales)
average_sales = total_sales / len(daily_sales)
highest_sales = max(daily_sales)
lowest_sales = min(daily_sales)

print("Total sales:", total_sales)
print("Average sales:", average_sales)
print("Highest sales:", highest_sales)
print("Lowest sales:", lowest_sales)

Total sales: 46450
Average sales: 3317.8571428571427
Highest sales: 5600
Lowest sales: 1950


In [29]:
# Find the positions of the best and worst sales days
best_index = daily_sales.index(highest_sales)
worst_index = daily_sales.index(lowest_sales)

# Use the positions to find the day names
best_day = days[best_index]
worst_day = days[worst_index]

# Positions 0-6 are Week 1, positions 7-13 are Week 2
best_week = 1 if best_index < 7 else 2
worst_week = 1 if worst_index < 7 else 2

print("Best day:", best_day, "- Week", best_week)
print("Worst day:", worst_day, "- Week", worst_week)

Best day: Sat - Week 2
Worst day: Tue - Week 1


## Task 2 — Weekly Totals and Growth

I will create a function called `week_totals()` that uses slicing to separate Week 1 and Week 2 sales. The function will return both weekly totals, and I will then calculate the percentage growth from Week 1 to Week 2.

In [30]:
def week_totals(sales):
    """Return the total sales for Week 1 and Week 2."""
    week1 = sum(sales[:7])
    week2 = sum(sales[7:])
    return week1, week2

In [31]:
week1_total, week2_total = week_totals(daily_sales)

growth_percentage = ((week2_total - week1_total) / week1_total) * 100

print("Week 1 total:", week1_total)
print("Week 2 total:", week2_total)
print("Week-on-week growth:", round(growth_percentage, 2), "%")

Week 1 total: 22050
Week 2 total: 24400
Week-on-week growth: 10.66 %


## Task 3 — Classify Daily Sales Performance

I will create a function called `classify_day()` to classify each day's sales as Strong, Normal, or Weak based on the average daily sales. I will then count how many days fall into each category.

In [32]:
def classify_day(amount, average):
    """Classify a day's sales as Strong, Normal, or Weak."""
    if amount > average * 1.20:
        return "Strong"
    elif amount < average * 0.80:
        return "Weak"
    else:
        return "Normal"

In [33]:
categories = {
    "Strong": 0,
    "Normal": 0,
    "Weak": 0
}

for sale in daily_sales:
    category = classify_day(sale, average_sales)
    categories[category] += 1

print("Daily performance categories:")
print(categories)

Daily performance categories:
{'Strong': 4, 'Normal': 4, 'Weak': 6}


## Task 4 — Weekday vs Weekend Sales

I will calculate the average sales for weekdays (Monday to Friday) and weekends (Saturday and Sunday), then calculate the percentage difference between them.

In [34]:
weekday_sales = []
weekend_sales = []

for i in range(len(daily_sales)):
    if days[i] in ["Mon", "Tue", "Wed", "Thu", "Fri"]:
        weekday_sales.append(daily_sales[i])
    else:
        weekend_sales.append(daily_sales[i])

weekday_average = sum(weekday_sales) / len(weekday_sales)
weekend_average = sum(weekend_sales) / len(weekend_sales)

print("Weekday average:", round(weekday_average, 2))
print("Weekend average:", round(weekend_average, 2))

Weekday average: 2685.0
Weekend average: 4900.0


In [35]:
percentage_difference = ((weekend_average - weekday_average) / weekday_average) * 100

print("Weekend sales are", round(percentage_difference, 2), "% higher than weekday sales.")

Weekend sales are 82.5 % higher than weekday sales.


## Task 5 — Stock Reordering and Gross Margin

I will identify products whose stock quantity is at or below their reorder level. I will also calculate the gross margin percentage for each product using its cost and selling price.

In [36]:
def products_to_reorder(stock_data):
    """Return a list of products whose quantity is at or below the reorder level."""
    reorder_list = []

    for product, details in stock_data.items():
        if details["qty"] <= details["reorder"]:
            reorder_list.append(product)

    return reorder_list

In [37]:
reorder_products = products_to_reorder(stock)

print("Products to reorder:", reorder_products)

Products to reorder: ['Tomatoes', 'Sukuma']


In [38]:
print("Gross Margin by Product:")

for product, details in stock.items():
    cost = details["cost"]
    price = details["price"]

    gross_margin = ((price - cost) / price) * 100

    print(product, ":", round(gross_margin, 2), "%")

Gross Margin by Product:
Tomatoes : 33.33 %
Onions : 30.0 %
Sukuma : 40.0 %
Potatoes : 33.33 %
Avocado : 42.86 %


## Task 6 — Stall Performance Report

I will print a formatted summary of the main sales, growth, daily performance, weekend performance, and stock findings using f-strings. KES amounts will be displayed with thousands separators for readability.

In [39]:
print("===== Stall Performance Report =====")

print(f"Total Sales: KES {total_sales:,.2f}")
print(f"Average Daily Sales: KES {average_sales:,.2f}")
print(f"Highest Daily Sales: KES {highest_sales:,.2f} ({best_day}, Week {best_week})")
print(f"Lowest Daily Sales: KES {lowest_sales:,.2f} ({worst_day}, Week {worst_week})")

print(f"Week 1 Total: KES {week1_total:,.2f}")
print(f"Week 2 Total: KES {week2_total:,.2f}")
print(f"Week-on-Week Growth: {growth_percentage:.2f}%")

print(f"Strong Days: {categories['Strong']}")
print(f"Normal Days: {categories['Normal']}")
print(f"Weak Days: {categories['Weak']}")

print(f"Weekday Average: KES {weekday_average:,.2f}")
print(f"Weekend Average: KES {weekend_average:,.2f}")
print(f"Weekend Sales Difference: {percentage_difference:.2f}% higher")

print(f"Products to Reorder: {', '.join(reorder_products)}")

===== Stall Performance Report =====
Total Sales: KES 46,450.00
Average Daily Sales: KES 3,317.86
Highest Daily Sales: KES 5,600.00 (Sat, Week 2)
Lowest Daily Sales: KES 1,950.00 (Tue, Week 1)
Week 1 Total: KES 22,050.00
Week 2 Total: KES 24,400.00
Week-on-Week Growth: 10.66%
Strong Days: 4
Normal Days: 4
Weak Days: 6
Weekday Average: KES 2,685.00
Weekend Average: KES 4,900.00
Weekend Sales Difference: 82.50% higher
Products to Reorder: Tomatoes, Sukuma


## Task 7 — Recommendations for Wanjiku

### Recommendations for Wanjiku

Wanjiku's sales increased from KES 22,050 in Week 1 to KES 24,400 in Week 2, representing a 10.66% growth, so the business is showing positive growth.

Weekend sales average KES 4,900 compared with KES 2,685 on weekdays, meaning weekends perform 82.50% better, so Wanjiku should ensure enough stock is available for Saturday and Sunday.

Tomatoes and Sukuma should be restocked because their current quantities are at or below their reorder levels.

Saturday of Week 2 was the best-performing day with KES 5,600 in sales, while Tuesday of Week 1 was the lowest at KES 1,950, suggesting Wanjiku should focus more stock and selling effort on stronger trading days.